cluster/app/demo/Main.hs

```diff
@@ -73,32 +73,20 @@ main = void $ do
         ]
 
     handles <- forM cluster $ \case
-        RunningCoreNode (NodeName nodeId) env handle -> do
-            putTextLn $ "..."
-                <> nodeId <> " has no health-check API."
-                <> "\n......system start:  "   <> toText (env ! "SYSTEM_START")
-                <> "\n......address:       "   <> toText (env ! "LISTEN")
-                <> "\n......locked assets: " <> maybe "-" toText ("ASSET_LOCK_FILE" `lookup` env)
-            return handle
-
-        RunningRelayNode (NodeName nodeId) env handle -> do
-            putTextLn $ "..."
-                <> nodeId <> " has no health-check API."
-                <> "\n......system start:  " <> toText (env ! "SYSTEM_START")
-                <> "\n......address:       " <> toText (env ! "LISTEN")
-            return handle
-
-        RunningEdgeNode (NodeName nodeId) env manager handle -> do
+        RunningNode nodeType (NodeName nodeId) env manager handle -> do
```

Comment:
The type has been simplified from an ADT with three constructor to just one constructor taking an extra `NodeType` argument. Cf below.